build.assets/Dockerfile-centos7-assets

```diff
@@ -36,22 +36,22 @@ RUN yum install --nogpgcheck -y \
     yum clean all
 
 # Use just created devtool image with newer GCC and Cmake
-FROM --platform=$BUILDPLATFORM centos-devtoolset as clang12
+FROM --platform=$BUILDPLATFORM centos-devtoolset as clang14
 
 ARG DEVTOOLSET
 
-# Compile Clang 12.0.0 from source. It is needed to create BoringSSL and BPF files.
+# Compile Clang 14.0.6 from source. It is needed to create BoringSSL and BPF files.
```

Comment:
Can you cut/publish a dev tag prior to merge? Jumping 2 major versions of clang could potentially cause build issues and/or cause unexpected bugs.

IMO this is major enough that we should go through the whole test plan with the version bumped, but that's probably not feasible at this stage.

Reply:
Dev tag won't actually test any of this, as the assets build is separate from the build pipeline and buildbox usage.

But that said, aren't we only using clang for BoringSSL and libbpf? Is there anything else that uses clang? I don't see why we would need to redo the entire test plan. At most, this would affect FIPS builds and BPF features.

Reply:
Ah alright. I was under the impression that we used Clang during much more of the build process. Do you know if we're actually shipping these built libbpf and boringssl libs to customers? If so then it'd still be nice to have some tests ran against the final build, but it's way lower risk than I originally thought.